File: C++ solutions/168. Excel Sheet Column Title.cpp
class Solution {
public:
    string convertToTitle(int columnNumber) {
        if (columnNumber < 27)
            return string(1, 'A' + (columnNumber - 1) % 26);
        
        string ans = "";
        while (columnNumber > 0) {
            if (columnNumber % 26 == 0) {
                ans += 'A' + 25;
                columnNumber -= 1;
            } else {
                ans += 'A' + columnNumber % 26 - 1;
            }
            columnNumber /= 26;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: JavaScript solutions/168. Excel Sheet Column Title.js
/**
 * @param {number} columnNumber
 * @return {string}
 */
var convertToTitle = function (columnNumber) {
  if (columnNumber < 27) {
    return String.fromCharCode("A".charCodeAt(0) + ((columnNumber - 1) % 26));
  }

  let ans = "";
  while (columnNumber > 0) {
    if (columnNumber % 26 === 0) {
      ans += String.fromCharCode("A".charCodeAt(0) + 25);
      columnNumber -= 1;
    } else {
      ans += String.fromCharCode("A".charCodeAt(0) + (columnNumber % 26) - 1);
    }
    columnNumber = Math.floor(columnNumber / 26);
  }

  return ans.split("").reverse().join("");
};
